tests/instruments/test_instrument.py

```diff
@@ -150,6 +151,41 @@ class Fake(FakeInstrument):
     assert fake.x == 5
 
 
+def test_control_preprocess_reply():
+    # test setting preprocess_reply at property-level
+    class Fake(FakeInstrument):
+        x = Instrument.control(
+            "", "JUNK%d", "",
+            preprocess_reply=lambda v: v.replace('JUNK', ''),
+            cast=int
+        )
+
+    fake = Fake()
+    fake.x = 5
+    assert fake.read() == 'JUNK5'
+    fake.x = 5
+    assert fake.x == 5
+    fake.x = 5
+    assert type(fake.x) == int
```

Comment:
I think the second fake.x is not needed here?
```suggestion
    assert fake.x == 5
    assert type(fake.x) == int
```

Reply:
Its a `FakeInstrument` with `FakeAdapter`. `FakeAdapter.read` seems to clear the buffer and therefore the second set is needed.
Here the corresponding error (of `python setup.py test` otherwise:
```
_______________________________ test_control_validator ________________________________

    def test_control_validator():
        class Fake(FakeInstrument):
            x = Instrument.control(
                "", "%d", "",
                validator=strict_discrete_set,
                values=range(10),
            )
    
        fake = Fake()
        fake.x = 5
        assert fake.read() == '5'
    #    fake.x = 5
>       assert fake.x == 5
E       AssertionError: assert '' == 5
E        +  where '' = <test_instrument.test_control_validator.<locals>.Fake object at 0x7f13a9713b50>.x

tests/instruments/test_instrument.py:62: AssertionError
```